packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2030,101 +2030,112 @@ mixin WidgetInspectorService {
+          // ignore: invalid_use_of_protected_member
+          final Constraints constraints = renderObject.constraints;
+          final Map<String, Object>constraintsProperty = <String, Object>{
+            'type': constraints.runtimeType.toString(),
+            'description': constraints.toString(),
+          };
+          if (constraints is BoxConstraints) {
+            constraintsProperty.addAll(<String, Object>{
+              'minWidth': constraints.minWidth.toString(),
+              'minHeight': constraints.minHeight.toString(),
+              'maxWidth': constraints.maxWidth.toString(),
+              'maxHeight': constraints.maxHeight.toString(),
+            });
+          }
+          additionalJson['constraints'] = constraintsProperty;
+        }
+      } catch (e) {
+        // Constraints are sometimes unavailable even though
+        // debugNeedsLayout is false.
+      }
+
+      try {
+        if (renderObject is RenderBox) {
+          additionalJson['isBox'] = true;
+          additionalJson['size'] = <String, Object>{
+            'width': renderObject.size.width.toString(),
+            'height': renderObject.size.height.toString(),
+          };
+
+          final ParentData? parentData = renderObject.parentData;
+          if (parentData is FlexParentData) {
+            additionalJson['flexFactor'] = parentData.flex!;
+            additionalJson['flexFit'] =
+                describeEnum(parentData.fit ?? FlexFit.tight);
+          } else if (parentData is BoxParentData) {
+            final Offset offset = parentData.offset;
+            additionalJson['parentData'] = <String, Object>{
+              'offsetX': offset.dx.toString(),
+              'offsetY': offset.dy.toString(),
+            };
+          }
+        } else if (renderObject is RenderView) {
+          additionalJson['size'] = <String, Object>{
+            'width': renderObject.size.width.toString(),
+            'height': renderObject.size.height.toString(),
+          };
+        }
+      } catch (e) {
+        // Not laid out yet.
+      }
+      return additionalJson;
+    }
+
     result = _nodeToJson(
       root as DiagnosticsNode,
       InspectorSerializationDelegate(
         groupName: groupName,
         summaryTree: true,
         subtreeDepth: subtreeDepth,
         service: this,
-        addAdditionalPropertiesCallback: (DiagnosticsNode node, InspectorSerializationDelegate delegate) {
-          final Map<String, Object> additionalJson = <String, Object>{};
-          final Object? value = node.value;
-          if (value is Element) {
-            final RenderObject? renderObject = value.renderObject;
-            if (renderObject != null) {
-              additionalJson['renderObject'] =
-                  renderObject.toDiagnosticsNode().toJsonMap(
-                        delegate.copyWith(
-                          subtreeDepth: 0,
-                          includeProperties: true,
```

Comment:
I think the simpler fix may be to add
expandProperties: false here
That should avoid the stack overflow without breaking the layout explorer.